WIP: EUI-4129: Activity Tracker

LucaDelBuonoHMCTS opened the pull request:
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4129

### Change description ###
WebSocket Activity Tracker

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[X] No
```

LucaDelBuonoHMCTS committed:
EUI-4129: Activity Tracker

LucaDelBuonoHMCTS committed:
EUI-4129: Some linting

LucaDelBuonoHMCTS committed:
EUI-4129: Update Toolkit version

LucaDelBuonoHMCTS committed:
EUI-4129: yarn audit

LucaDelBuonoHMCTS committed:
EUI-4129: Fix tests

LucaDelBuonoHMCTS committed:
EUI-4129: Update SERVICES_CCD_ACTIVITY_API in values.preview.template.yaml

LucaDelBuonoHMCTS committed:
EUI-4129: Yarn audit

LucaDelBuonoHMCTS committed:
EUI-4129: Update env variables

LucaDelBuonoHMCTS committed:
EUI-4129: Replace @hmcts/http-proxy-middleware with http-proxy-middleware

LucaDelBuonoHMCTS committed:
EUI-4129: Replace SERVICES_CCD_ACTIVITY_API_PATH value

LucaDelBuonoHMCTS committed:
Merge branch 'master' into feature/EUI-4129-Activity-Tracker

# Conflicts:
#	package.json
#	yarn.lock

LucaDelBuonoHMCTS committed:
EUI-4129: Merge master into branch

LucaDelBuonoHMCTS committed:
EUI-4129: Linting

LucaDelBuonoHMCTS committed:
EUI-4129: Reorder code

LucaDelBuonoHMCTS committed:
EUI-4129: Reorder code

LucaDelBuonoHMCTS committed:
EUI-4129: Reorder code and add back rewriteUrl

LucaDelBuonoHMCTS committed:
EUI-4129: /activity endpoint as a ws

LucaDelBuonoHMCTS committed:
EUI-4129: Revert /activity endpoint as a ws

LucaDelBuonoHMCTS committed:
EUI-4129: Remove unused proxies

LucaDelBuonoHMCTS committed:
EUI-4129: Remove unused proxies

LucaDelBuonoHMCTS committed:
EUI-4129: Use CCD Gateway for normal polling

hmcts-jenkins-j-to-z[bot] committed:
Bumping chart version/ fixing aliases

LucaDelBuonoHMCTS committed:
EUI-4129: Remove unused import

LucaDelBuonoHMCTS committed:
Merge branch 'feature/EUI-4129-Activity-Tracker' of github.com:hmcts/rpx-xui-webapp into feature/EUI-4129-Activity-Tracker

dean-bullock reviewed (dismissed):
Approved. Lots of formatting changes make the diff quite noisy. Something to look at getting everyone's code editor formatters in-line

Mo-Lala84 reviewed (dismissed):
approved

LucaDelBuonoHMCTS committed:
Merge branch 'master' into feature/EUI-4129-Activity-Tracker

LucaDelBuonoHMCTS committed:
Merge branch 'master' into feature/EUI-4129-Activity-Tracker

# Conflicts:
#	package.json
#	yarn.lock

LucaDelBuonoHMCTS committed:
EUI-4129: Merge master into branch

LucaDelBuonoHMCTS committed:
Merge branch 'master' into feature/EUI-4129-Activity-Tracker

# Conflicts:
#	package.json
#	yarn.lock

LucaDelBuonoHMCTS committed:
Merge master into branch

LucaDelBuonoHMCTS committed:
Merge branch 'master' into feature/EUI-4129-Activity-Tracker

# Conflicts:
#	api/configuration/index.ts
#	package.json
#	yarn.lock

LucaDelBuonoHMCTS committed:
Merge master into branch

LucaDelBuonoHMCTS committed:
EUI-4129: Fix pagination property names

LucaDelBuonoHMCTS committed:
EUI-4129: yarn audit

LucaDelBuonoHMCTS committed:
EUI-4129: Sonar ignore

LucaDelBuonoHMCTS committed:
EUI-4129: Update Toolkit version

LucaDelBuonoHMCTS committed:
Merge branch 'master' into feature/EUI-4129-Activity-Tracker

# Conflicts:
#	api/application.ts
#	api/lib/middleware/proxy.ts
#	api/proxy.config.ts
#	package.json
#	src/cases/containers/case-list/case-list.component.spec.ts
#	src/cases/containers/case-list/case-list.component.ts
#	src/cases/containers/case-search/case-search.component.ts
#	src/cases/resolvers/activity.resolver.spec.ts
#	src/cases/resolvers/activity.resolver.ts
#	src/cases/resolvers/create-case-event-trigger.resolver.spec.ts
#	yarn.lock

LucaDelBuonoHMCTS committed:
EUI-4129: Linting after merge from master

LucaDelBuonoHMCTS committed:
Merge master into branch

LucaDelBuonoHMCTS committed:
Merge branch 'master' into feature/EUI-4129-Activity-Tracker

# Conflicts:
#	yarn-audit-known-issues

LucaDelBuonoHMCTS committed:
Merge master into branch

bradley-hmcts committed:
Merge branch 'master' into feature/EUI-4129-Activity-Tracker

LucaDelBuonoHMCTS committed:
Merge branch 'master' into feature/EUI-4129-Activity-Tracker

# Conflicts:
#	package.json
#	sonar-project.properties
#	src/cases/store/reducers/case-list.reducer.ts
#	src/cases/store/reducers/search-filter.reducer.ts
#	yarn.lock

LucaDelBuonoHMCTS committed:
EUI-4129: Merge master into branch

LucaDelBuonoHMCTS committed:
EUI-4129: Fix tests

LucaDelBuonoHMCTS committed:
EUI-4129: Remove test for LaunchDarkly id

LucaDelBuonoHMCTS committed:
Merge branch 'master' into feature/EUI-4129-Activity-Tracker

# Conflicts:
#	package.json
#	test/integration/tests/nodeApp_endpoints.spec.ts
#	test_codecept/integration/tests/nodeApp_endpoints.spec.ts
#	yarn.lock

LucaDelBuonoHMCTS committed:
EUI-4129: Merge master into branch

andywilkinshmcts committed:
Merge branch 'master' into feature/EUI-4129-Activity-Tracker

# Conflicts:
#	api/application.ts
#	api/proxy.config.ts
#	charts/xui-webapp/Chart.yaml
#	package.json
#	proxy.config.json
#	src/app/app.constants.ts
#	src/app/services/ccd-config/ccd-case.config.ts
#	src/cases/resolvers/activity.resolver.ts
#	test/integration/tests/nodeApp_endpoints.spec.ts
#	yarn.lock

andywilkinshmcts committed:
Fix merge issues

hmcts-jenkins-j-to-z[bot] committed:
Bumping chart version/ fixing aliases

balajisridharanhmcts committed:
merge latest master

hmcts-jenkins-j-to-z[bot] committed:
Bumping chart version/ fixing aliases

balajisridharanhmcts committed:
linting conflicts fixed

balajisridharanhmcts committed:
conflicts fixed

balajisridharanhmcts committed:
yarn lint issues

balajisridharanhmcts committed:
yarn lint issues

balajisridharanhmcts committed:
yarn lint issues

balajisridharanhmcts committed:
yarn lint issues

balajisridharanhmcts committed:
lint issue temp fix

balajisridharanhmcts committed:
lint issue fix

kiran-yenigala-hmcts committed:
CME-782: Angular 20 upgrade for MC

kiran-yenigala-hmcts committed:
Merge branch 'master' into CME-782

kiran-yenigala-hmcts committed:
CME-782: Angular 20 upgrade for MC

kiran-yenigala-hmcts committed:
CME-782: Fix lint issues

kiran-yenigala-hmcts committed:
CME-782: update version of rpx-xui-common-lib

kiran-yenigala-hmcts committed:
Merge branch 'master' into CME-782

kiran-yenigala-hmcts committed:
CME-782: update version of rpx-xui-common-lib

kiran-yenigala-hmcts committed:
CME-782: update version of rpx-xui-common-lib

kiran-yenigala-hmcts committed:
Merge branch 'master' into CME-782

kiran-yenigala-hmcts committed:
CME-782: fix pact test failures

kiran-yenigala-hmcts committed:
CME-782: fix pact test failures

kiran-yenigala-hmcts committed:
CME-782: fix pact test failures

kiran-yenigala-hmcts committed:
CME-782: revert to previous version

kiran-yenigala-hmcts committed:
Merge branch 'master' into CME-782

kiran-yenigala-hmcts committed:
CME-782: update rpx-xui-common-lib to 3.1.3-cme-778-prerelease

kiran-yenigala-hmcts committed:
CME-782: update rpx-xui-common-lib to 3.1.3-cme-778-prerelease

anthonydummer committed:
Update with release versions

anthonydummer committed:
Merge branch 'master' into CME-782

# Conflicts:
#	package.json
#	yarn.lock

anthonydummer committed:
Merge branch 'master' into CME-782

anthonydummer committed:
Merge branch 'master' into CME-782

anthonydummer committed:
update of ngrx

anthonydummer committed:
Merge branch 'master' into CME-782

# Conflicts:
#	package.json
#	yarn.lock

anthonydummer committed:
Merge branch 'master' into CME-782

AntonyLeons committed:
style: remove empty lines and reorder imports in test files

Clean up test files by removing unnecessary empty lines and alphabetizing imports for better readability and consistency

AntonyLeons committed:
style: remove empty lines in component decorators

Clean up component files by removing unnecessary empty lines in @Component decorators across multiple files to improve code consistency and readability

AntonyLeons committed:
style: remove empty lines from component decorators

AntonyLeons committed:
style: remove empty lines in component decorators across test files

AntonyLeons committed:
style: remove empty lines from component decorators in multiple files

AntonyLeons committed:
style: remove trailing whitespace from component files

AntonyLeons committed:
test(documents): update test imports and function references

Update test file to use direct import of handleRequest instead of namespace import
Remove unused mocha import and clean up test cases

AntonyLeons committed:
refactor(tests): clean up test imports and setup

Remove unused imports and consolidate chai setup using 'use' instead of 'chai.use' for better readability and consistency across test files

AntonyLeons committed:
style: remove unnecessary blank lines in test components

AntonyLeons committed:
chore: update angular-eslint packages and @types/node to latest versions

- Upgrade all @angular-eslint packages from 17.5.3 to 20.4.0
- Update @types/node from ^18.17.0 to ^20.19.0
- Update related dependencies to maintain compatibility

AntonyLeons committed:
Update .dockerignore

balajisridharanhmcts committed:
toolkit library added

balajisridharanhmcts committed:
Merge remote-tracking branch 'origin/master' into feature/EUI-4129-Activity-Tracker

hmcts-jenkins-j-to-z[bot] committed:
Bumping chart version/ fixing aliases

balajisridharanhmcts committed:
yarn audit fixes

balajisridharanhmcts committed:
functional test fix

AntonyLeons committed:
chore: update @hmcts/ccd-case-ui-toolkit to version 7.3.4-angular-20

AntonyLeons committed:
Update yarn-audit-known-issues

AntonyLeons committed:
build: update @hmcts/rpx-xui-common-lib dependency to 3.1.5-cme-778-prerelease

anthonydummer committed:
Post SRT merge

anthonydummer committed:
trailing comma

anthonydummer committed:
resolve merge conflicts

anthonydummer committed:
Update of ccd-case-ui-toolkit

anthonydummer committed:
update or yarn.lock

kiran-yenigala-hmcts committed:
CME-782: update ccpay-web-component version

kiran-yenigala-hmcts committed:
CME-782: Fix build issues

Josh-HMCTS committed:
fix smoke test step

kiran-yenigala-hmcts committed:
CME-782: update ccd-case-ui-toolkit version

kiran-yenigala-hmcts committed:
CME-782: update ccd-case-ui-toolkit version

Josh-HMCTS committed:
Merge branch 'master' into CME-782

Josh-HMCTS committed:
Merge branch 'master' into CME-782

kiran-yenigala-hmcts committed:
Merge branch 'master' into CME-782

balajisridharanhmcts committed:
Merge remote-tracking branch 'origin/CME-782' into feature/EUI-4129-Activity-Tracker

balajisridharanhmcts committed:
angular 20 upgrade merged

hmcts-jenkins-j-to-z[bot] committed:
Bumping chart version/ fixing aliases

balajisridharanhmcts committed:
toolkit version updated

balajisridharanhmcts committed:
unit test fixed

balajisridharanhmcts committed:
lint fixes

balajisridharanhmcts committed:
smoke test update

connorpgpmcelroy commented on api/application.ts:
```diff
@@ -58,6 +58,76 @@ export async function createApp() {
+    app.use(helmet.hidePoweredBy());
+    app.use(helmet.hsts({ maxAge: 28800000 }));
+    app.use(helmet.xssFilter());
+    app.use(helmet.contentSecurityPolicy({
```
This is no longer needed and should not be necessary - you should re-evaluate this file because I don't think any of the above is necessary unless it is relevant for activity tracker (which I don't think it is)

connorpgpmcelroy commented on src/cases/resolvers/activity.resolver.ts:
```diff
@@ -1,18 +1,28 @@
-      console.log(err);
-    }
-    return of(true);
+    return of(this.activityService.mode !== ActivityService.MODES.off);
```
This boolean is referred to twice - should be a boolean variable

connorpgpmcelroy replied:
Hasn't actually been replaced here though right? Should be return of(this.isActivityEnabled())?

connorpgpmcelroy reviewed (changes requested):
Needs a few changes

balajisridharanhmcts committed:
Merge remote-tracking branch 'origin/master' into feature/EUI-4129-Activity-Tracker

balajisridharanhmcts committed:
comments addressed

balajisridharanhmcts commented on src/work-allocation/components/case-name-field/case-name-field.component.ts:
```diff
@@ -21,10 +22,10 @@ export class CaseNameFieldComponent implements OnChanges {
-      const caseTypeId = this.caseType;
-      const caseId = this.caseId;
-      if (caseId && jurisdictionId && caseTypeId) {
+      if (this.caseId && this.jurisdiction && this.caseType) {
```
Done

balajisridharanhmcts committed:
yarn lint fix

connorpgpmcelroy commented on src/work-allocation/components/case-reference-field/case-reference-field.component.ts:
```diff
@@ -19,13 +19,16 @@ export class CaseReferenceFieldComponent implements OnChanges {
-        const jurisdictionId = encodeURIComponent(this.jurisdiction);
-        const caseTypeId = encodeURIComponent(this.caseType);
-        href = `${AppConstants.CASE_DETAILS_URL}${jurisdictionId}/${caseTypeId}/${caseId}`;
+      if (caseReference && jurisdiction && caseType) {
```
Same as above, don't need two ifs here

connorpgpmcelroy reviewed (dismissed):
Approving because my concerns are now minor and I won't be here for the second half of the day to change my review, however would ideally like those comments addressed

balajisridharanhmcts committed:
condition chk updates

balajisridharanhmcts committed:
temporary func test skip

balajisridharanhmcts committed:
temp skip few func test

balajisridharanhmcts committed:
temp skip few func test

balajisridharanhmcts committed:
API env var name change

balajisridharanhmcts committed:
ccd env url update

Josh-HMCTS committed:
Updates

Josh-HMCTS committed:
set the iss on preview

Josh-HMCTS committed:
update the value for activity api on preview

balajisridharanhmcts committed:
toolkit version update solicitor fix

balajisridharanhmcts committed:
toolkit version update

balajisridharanhmcts committed:
toolkit version upgrade

balajisridharanhmcts committed:
audit update version fix
